graphdatascience/aura_graph_data_science.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Dict, Optional, Tuple
+
+from neo4j import GraphDatabase
+from pandas import DataFrame
+
+from .query_runner.neo4j_query_runner import Neo4jQueryRunner
+from .server_version.server_version import ServerVersion
+from graphdatascience import GraphDataScience
+from graphdatascience.query_runner.aura_db_arrow_query_runner import (
+    AuraDbArrowQueryRunner,
+    AuraDbConnectionInfo,
+)
+
+
+class AuraGraphDataScience(GraphDataScience):
```

Comment:
Did we consider to compose instead of inherit? We do a lot of inheritance in the client, but maybe this is a place where we would be less limited if we didn't have to support all the same signatures?

Reply:
We did consider it, however we need the `Neo4jQueryRunner` and the `ArrowQueryRunner` from the `GraphDataScience` object. If we copy the code we basically copy almost the whole constructor. If we instantiate a gds object instead of extending it we still have to provide the same arguments.
I'm also not super happy with the initalization right now but is was the easiest for now. One other thing would be to disassemble the `GraphDataScience` constructor a bit and pulling things into own classes that we could reuse in the `AuraGraphDataScience` object.

Reply:
See #539

Reply:
wow, I love it!